From 023989f2374e1f0a3aa02faa27db9b8d825b06c5 Mon Sep 17 00:00:00 2001
From: Michael W <baddaemon87@gmail.com>
Date: Sat, 22 Oct 2016 14:47:09 +0200
Subject: [PATCH 095/211] Snap: Fix possible NPE

Seems like getSupportedFocusModes() can return null.
Check for null before using it further

Change-Id: I46860f780b99970cbe9857c8ba73691d69321488
Reference: BugDumps 20161014-20 L#131
---
 src/com/android/camera/PhotoModule.java | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/com/android/camera/PhotoModule.java b/src/com/android/camera/PhotoModule.java
index 6782f8bd2..fdd768453 100644
--- a/src/com/android/camera/PhotoModule.java
+++ b/src/com/android/camera/PhotoModule.java
@@ -5064,8 +5064,10 @@ public class PhotoModule
         mAeLockSupported = CameraUtil.isAutoExposureLockSupported(mInitialParams);
         mAwbLockSupported = CameraUtil.isAutoWhiteBalanceLockSupported(mInitialParams);
 
-        if (mInitialParams.getSupportedFocusModes().contains(CameraUtil.FOCUS_MODE_CONTINUOUS_PICTURE) ||
-            mInitialParams.getSupportedFocusModes().contains(CameraUtil.FOCUS_MODE_MW_CONTINUOUS_PICTURE)) {
+        List<String> focusModes = mInitialParams.getSupportedFocusModes();
+        if (focusModes != null &&
+                (focusModes.contains(CameraUtil.FOCUS_MODE_CONTINUOUS_PICTURE) ||
+                focusModes.contains(CameraUtil.FOCUS_MODE_MW_CONTINUOUS_PICTURE))) {
             mContinuousFocusSupported = true;
         } else {
             mContinuousFocusSupported = false;
-- 
2.17.1

